src/pages/Postcreatepage/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import  { Redirect } from 'react-router-dom';
+import { Form } from 'react-bootstrap';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import { createPost } from 'store/actions/createPost';
+import { Styled } from 'pages/Postcreatepage/Postcreatepage.styles';
+import CommonLayout from 'layout/CommonLayout';
+import CustomInputComponent from 'components/Input';
+import CustomButton from 'components/Button';
+
+
+const Postcreatepage = ( props: any ) => {
+    const [redirect, setRedirect] = useState<boolean>(false);
+
+    const submitHandler = (values: any) => {
+        const post = {
+            ...values,
+            owner: localStorage.getItem('userName')
+        }
+        props.createPostHandler(post.title, post.place, post.description, post.owner);
+        setRedirect(true);
+    };
+
+    return(
+        <CommonLayout>
+            {redirect ? <Redirect to='/homepage' /> : null}
+            <Styled.InnerContainer>
+                <Styled.Header>Create post</Styled.Header>
+                <Formik
+                    initialValues={{title:'', description: '', place: ''}}
+                    validationSchema={Yup.object({
+                        title: Yup.string().required('Required'),
+                        description: Yup.string().required('Required'),
+                        place: Yup.string().required('Required')
+                    })}
+                    onSubmit={(values) =>{submitHandler(values)}}
+                    >
```

Comment:
Change it to useFormik hook to make the tsx portion cleaner.

Reply:
Used useFormik hook.